src/Simplify.elm

```diff
@@ -6981,6 +6991,64 @@ stringDetermineLength expression =
             Nothing
 
 
+arrayCollection : CollectionProperties {}
+arrayCollection =
+    { moduleName = [ "Array" ]
+    , represents = "array"
+    , empty =
+        { description = Constant (qualifiedToString ( [ "Array" ], "empty" ))
+        , is =
+            \lookupTable expr ->
+                isJust (AstHelpers.getSpecificValueOrFunction ( [ "Array" ], "empty" ) lookupTable expr)
+        , asString =
+            \resources ->
+                qualifiedToString (qualify ( [ "Array" ], "empty" ) resources)
+        }
+    , nameForSize = "length"
+    , determineSize = arrayDetermineSize
+    }
+
+
+arrayDetermineSize :
+    ModuleNameLookupTable
+    -> Node Expression
+    -> Maybe CollectionSize
+arrayDetermineSize lookupTable expressionNode =
+    firstThatConstructsJust
+        [ \() ->
+            case AstHelpers.getSpecificValueOrFunction ( [ "Array" ], "empty" ) lookupTable expressionNode of
+                Just _ ->
+                    Just (Exactly 0)
+
+                Nothing ->
+                    Nothing
+        , \() ->
+            case AstHelpers.getSpecificFunctionCall ( [ "Array" ], "fromList" ) lookupTable expressionNode of
+                Just fromListCall ->
+                    case AstHelpers.getListLiteral fromListCall.firstArg of
+                        Just [] ->
+                            Just (Exactly 0)
+
+                        Just (_ :: []) ->
+                            Just (Exactly 1)
+
+                        Just (el0 :: el1 :: el2Up) ->
+                            case traverse getComparableExpression (el0 :: el1 :: el2Up) of
```

Comment:
I think you copied this from set or somewhere. comparable etc certainly doesn't apply here.
I'd just use `listDetermineLength` for the list from `Array.fromList`

Reply:
:+1: